pkg/kubelet/qos/policy_test.go

```diff
@@ -159,81 +171,270 @@ var (
+				v1.ResourceName(v1.ResourceMemory): sampleDefaultMemLimit,
+			},
+		},
+	}
+	restartPolicyAlways    = v1.ContainerRestartPolicyAlways
+	sampleSidecarContainer = v1.Container{
+		Name:          "sidecar-container",
+		RestartPolicy: &restartPolicyAlways,
+		Resources: v1.ResourceRequirements{
+			Requests: v1.ResourceList{
+				v1.ResourceName(v1.ResourceMemory): sampleDefaultMemRequest,
+			},
+			Limits: v1.ResourceList{
+				v1.ResourceName(v1.ResourceMemory): sampleDefaultMemLimit,
+			},
+		},
+	}
+
+	sampleSmallSidecarContainer = v1.Container{
+		Name:          "sidecar-small-container",
+		RestartPolicy: &restartPolicyAlways,
+		Resources: v1.ResourceRequirements{
+			Requests: v1.ResourceList{
+				v1.ResourceName(v1.ResourceMemory): resource.MustParse(strconv.FormatInt(standardMemoryAmount/20, 10)),
+			},
+			Limits: v1.ResourceList{
+				v1.ResourceName(v1.ResourceMemory): sampleDefaultMemLimit,
+			},
+		},
+	}
+
+	sampleBigSidecarContainer = v1.Container{
+		Name:          "sidecar-big-container",
+		RestartPolicy: &restartPolicyAlways,
+		Resources: v1.ResourceRequirements{
+			Requests: v1.ResourceList{
+				v1.ResourceName(v1.ResourceMemory): resource.MustParse(strconv.FormatInt(standardMemoryAmount/2, 10)),
+			},
+			Limits: v1.ResourceList{
+				v1.ResourceName(v1.ResourceMemory): sampleDefaultMemLimit,
+			},
+		},
+	}
+
+	burstableMixedUniqueMainContainerPod = v1.Pod{
+		Spec: v1.PodSpec{
+			InitContainers: []v1.Container{
+				sampleInitContainer,
+			}, Containers: []v1.Container{
+				sampleContainer,
+			},
+		},
+	}
+
+	burstableMixedMultiContainerSameRequestPod = v1.Pod{
+		Spec: v1.PodSpec{
+			InitContainers: []v1.Container{
+				sampleInitContainer, sampleSidecarContainer,
+			}, Containers: []v1.Container{
+				sampleContainer,
+			},
+		},
+	}
+
+	burstableMixedMultiContainerSmallSideCarePod = v1.Pod{
+		Spec: v1.PodSpec{
+			InitContainers: []v1.Container{
+				sampleInitContainer, sampleSmallSidecarContainer,
+			}, Containers: []v1.Container{
+				sampleContainer,
+			},
+		},
+	}
+
+	burstableMixedMultiContainerBigSidecarContainerPod = v1.Pod{
+		Spec: v1.PodSpec{
+			InitContainers: []v1.Container{
+				sampleInitContainer, sampleBigSidecarContainer,
+			}, Containers: []v1.Container{
+				sampleContainer,
+			},
+		},
+	}
 )
 
+type lowHighOOMScoreAdjTest struct {
+	lowOOMScoreAdj  int
+	highOOMScoreAdj int
+}
 type oomTest struct {
-	pod             *v1.Pod
-	memoryCapacity  int64
-	lowOOMScoreAdj  int // The max oom_score_adj score the container should be assigned.
-	highOOMScoreAdj int // The min oom_score_adj score the container should be assigned.
+	pod                             *v1.Pod
+	memoryCapacity                  int64
+	lowHighOOMScoreAdj              map[string]lowHighOOMScoreAdjTest // [container-name] : min and max oom_score_adj score the container should be assigned.
+	sidecarContainersFeatureEnabled bool
 }
 
 func TestGetContainerOOMScoreAdjust(t *testing.T) {
```

Comment:
Since it's a little hard to see which test failed, could you add the test name? Something like this.

```diff
diff --git a/pkg/kubelet/qos/policy_test.go b/pkg/kubelet/qos/policy_test.go
index 90c4cae5628..862add6ecc0 100644
--- a/pkg/kubelet/qos/policy_test.go
+++ b/pkg/kubelet/qos/policy_test.go
@@ -314,71 +314,71 @@ type oomTest struct {
 }

 func TestGetContainerOOMScoreAdjust(t *testing.T) {
-       oomTests := []oomTest{
-               {
+       oomTests := map[string]oomTest{
+               "cpu-limit": {
                        pod:            &cpuLimit,
                        memoryCapacity: 4000000000,
                        lowHighOOMScoreAdj: map[string]lowHighOOMScoreAdjTest{
                                "cpu-limit": {lowOOMScoreAdj: 999, highOOMScoreAdj: 999},
                        },
                },
-               {
+               "memory-limit-cpu-request": {
                        pod:            &memoryLimitCPURequest,
                        memoryCapacity: 8000000000,
                        lowHighOOMScoreAdj: map[string]lowHighOOMScoreAdjTest{
                                "memory-limit-cpu-request": {lowOOMScoreAdj: 999, highOOMScoreAdj: 999},
                        },
                },
-               {
+               "zero-memory-limit": {
                        pod:            &zeroMemoryLimit,
                        memoryCapacity: 7230457451,
                        lowHighOOMScoreAdj: map[string]lowHighOOMScoreAdjTest{
                                "zero-memory-limit": {lowOOMScoreAdj: 1000, highOOMScoreAdj: 1000},
                        },
                },
-               {
+               "no-request-limit": {
                        pod:            &noRequestLimit,
                        memoryCapacity: 4000000000,
                        lowHighOOMScoreAdj: map[string]lowHighOOMScoreAdjTest{
                                "no-request-limit": {lowOOMScoreAdj: 1000, highOOMScoreAdj: 1000},
                        },
                },
-               {
+               "equal-request-limit-cpu-memory": {
                        pod:            &equalRequestLimitCPUMemory,
                        memoryCapacity: 123456789,
                        lowHighOOMScoreAdj: map[string]lowHighOOMScoreAdjTest{
                                "equal-request-limit-cpu-memory": {lowOOMScoreAdj: -997, highOOMScoreAdj: -997},
                        },
                },
-               {
+               "cpu-unlimited-memory-limited-with-requests": {
                        pod:            &cpuUnlimitedMemoryLimitedWithRequests,
                        memoryCapacity: standardMemoryAmount,
                        lowHighOOMScoreAdj: map[string]lowHighOOMScoreAdjTest{
                                "cpu-unlimited-memory-limited-with-requests": {lowOOMScoreAdj: 495, highOOMScoreAdj: 505},
                        },
                },
-               {
+               "request-no-limit": {
                        pod:            &requestNoLimit,
                        memoryCapacity: standardMemoryAmount,
                        lowHighOOMScoreAdj: map[string]lowHighOOMScoreAdjTest{
                                "request-no-limit": {lowOOMScoreAdj: 3, highOOMScoreAdj: 3},
                        },
                },
-               {
+               "cluster-critical": {
                        pod:            &clusterCritical,
                        memoryCapacity: 4000000000,
                        lowHighOOMScoreAdj: map[string]lowHighOOMScoreAdjTest{
                                "cluster-critical": {lowOOMScoreAdj: 1000, highOOMScoreAdj: 1000},
                        },
                },
-               {
+               "node-critical": {
                        pod:            &nodeCritical,
                        memoryCapacity: 4000000000,
                        lowHighOOMScoreAdj: map[string]lowHighOOMScoreAdjTest{
                                "node-critical": {lowOOMScoreAdj: -997, highOOMScoreAdj: -997},
                        },
                },
-               {
+               "burstable-unique-container-pod": {
                        pod:            &burstableUniqueContainerPod,
                        memoryCapacity: standardMemoryAmount,
                        lowHighOOMScoreAdj: map[string]lowHighOOMScoreAdjTest{
@@ -386,7 +386,7 @@ func TestGetContainerOOMScoreAdjust(t *testing.T) {
                        },
                        sidecarContainersFeatureEnabled: true,
                },
-               {
+               "burstable-mixed-unique-main-container-pod": {
                        pod:            &burstableMixedUniqueMainContainerPod,
                        memoryCapacity: standardMemoryAmount,
                        lowHighOOMScoreAdj: map[string]lowHighOOMScoreAdjTest{
@@ -395,17 +395,17 @@ func TestGetContainerOOMScoreAdjust(t *testing.T) {
                        },
                        sidecarContainersFeatureEnabled: true,
                },
-               {
+               "burstable-mixed-multi-container-small-sidecar-pod": {
                        pod:            &burstableMixedMultiContainerSmallSideCarePod,
                        memoryCapacity: standardMemoryAmount,
                        lowHighOOMScoreAdj: map[string]lowHighOOMScoreAdjTest{
                                "init-container":          {lowOOMScoreAdj: 875, highOOMScoreAdj: 880},
                                "sidecar-small-container": {lowOOMScoreAdj: 951, highOOMScoreAdj: 951}, // first to kill
-                               "main-1":                  {lowOOMScoreAdj: 875, highOOMScoreAdj: 875},
+                               "main-1":                  {lowOOMScoreAdj: 875, highOOMScoreAdj: 880},
                        },
                        sidecarContainersFeatureEnabled: true,
                },
-               {
+               "burstable-mixed-multi-container-sample-request-pod": {
                        pod:            &burstableMixedMultiContainerSameRequestPod,
                        memoryCapacity: standardMemoryAmount,
                        lowHighOOMScoreAdj: map[string]lowHighOOMScoreAdjTest{
@@ -415,7 +415,7 @@ func TestGetContainerOOMScoreAdjust(t *testing.T) {
                        },
                        sidecarContainersFeatureEnabled: true,
                },
-               {
+               "burstable-mixed-multi-container-big-sidecar-container-pod": {
                        pod:            &burstableMixedMultiContainerBigSidecarContainerPod,
                        memoryCapacity: standardMemoryAmount,
                        lowHighOOMScoreAdj: map[string]lowHighOOMScoreAdjTest{
@@ -426,15 +426,18 @@ func TestGetContainerOOMScoreAdjust(t *testing.T) {
                        sidecarContainersFeatureEnabled: true,
                },
        }
-       for _, test := range oomTests {
-               featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.SidecarContainers, test.sidecarContainersFeatureEnabled)
-               listContainers := test.pod.Spec.InitContainers
-               listContainers = append(listContainers, test.pod.Spec.Containers...)
-               for _, container := range listContainers {
-                       oomScoreAdj := GetContainerOOMScoreAdjust(test.pod, &container, test.memoryCapacity)
-                       if oomScoreAdj < test.lowHighOOMScoreAdj[container.Name].lowOOMScoreAdj || oomScoreAdj > test.lowHighOOMScoreAdj[container.Name].highOOMScoreAdj {
-                               t.Errorf("oom_score_adj %s should be between %d and %d, but was %d", container.Name, test.lowHighOOMScoreAdj[container.Name].lowOOMScoreAdj, test.lowHighOOMScoreAdj[container.Name].highOOMScoreAdj, oomScoreAdj)
+       for name, test := range oomTests {
+               t.Run(name, func(t *testing.T) {
+                       featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.SidecarContainers, test.sidecarContainersFeatureEnabled)
+                       listContainers := test.pod.Spec.InitContainers
+                       listContainers = append(listContainers, test.pod.Spec.Containers...)
+                       for _, container := range listContainers {
+                               oomScoreAdj := GetContainerOOMScoreAdjust(test.pod, &container, test.memoryCapacity)
+                               if oomScoreAdj < test.lowHighOOMScoreAdj[container.Name].lowOOMScoreAdj || oomScoreAdj > test.lowHighOOMScoreAdj[container.Name].highOOMScoreAdj {
+                                       t.Errorf("oom_score_adj %s should be between %d and %d, but was %d", container.Name, test.lowHighOOMScoreAdj[container.Name].lowOOMScoreAdj, test.lowHighOOMScoreAdj[container.Name].highOOMScoreAdj, oomScoreAdj)
+                               }
+                               t.Logf("[%s]: oomScoreAdj: %d", container.Name, oomScoreAdj)
                        }
-               }
+               })
        }
 }
```

Reply:
Done